luaui/barwidgets.lua

```diff
@@ -920,19 +951,33 @@ function widgetHandler:EnableWidget(name, enableLocalsAccess)
 		end
 		self:InsertWidget(w)
 		self:SaveConfigData()
+
+		-- re-enable child widgets if they were previously enabled
+		if ki.children and #ki.children > 0 then
+			for _, childInfo in ipairs(ki.children) do
+				local active = self.knownWidgetInfos[childInfo.name].active
+				local enabled = self.orderList[childInfo.name] and (self.orderList[childInfo.name] > 0)
+
+				if not active and enabled then
+					self:EnableWidget(childInfo.name, enableLocalsAccess)
+				end
+			end
+		end
 	end
 	return true
 end
 
 function widgetHandler:DisableWidget(name)
-	local ki = self.knownWidgets[name]
+	local ki = self.knownWidgetInfos[name]
 	if not ki then
 		Spring.Echo("DisableWidget(), could not find widget: " .. tostring(name))
 		return false
 	end
 	if ki.active then
 		local w = self:FindWidget(name)
 		if not w then
+			Spring.Echo("DisableWidget(), found active info but could not find widget: " .. tostring(name))
+			Spring.Debug.TableEcho(ki)
```

Comment:
probably remove the tableEcho.

I did actually trigger this somehow (the screenshot I added was the initial state), but now I can't get back to that state :(